gooey/gui/components/widgets/richtextconsole.py

```diff
@@ -59,6 +61,27 @@ def __init__(self, parent):
             # NB : we use a default parameter to force the evaluation of the binding
             self.actionsMap[escSeq] = lambda bindedColor=wxcolor: self.BeginTextColour(bindedColor)
 
+    def AppendText2(self, content):
+        """Capture URLs and make them blue, underlined, and clickable."""
+        while True:
```

Comment:
I don't dig this while loop + mutation of the content variable. 

As a possible alternative, if you expand your regex's capture group to include the whole match, you can simplify this whole thing to a single forloop + if statement

```python 
url_pattern = r'\b((?:file|https?)://[^\s<>]*)'
```

>note that we treat the inner scheme subgroup as non-capturing 

then re.split will give you stable pairs of [plaintext, url]

```python
re.split(r, 'http://www.example.com')
>>> ['', 'http://www.example.com', '']
re.split(r, 'Hello world!\n\nhttp://www.example.com')
>>> ['Hello world!\n\n', 'http://www.example.com', '']
re.split(r, 'file://www.example.com')
>>> ['', 'file://www.example.com', '']
re.split(r, 'some text file://www.example.com some more text http://www.example.com http://www.example.com')
>>> ['some text ', 'file://www.example.com', ' some more text ', 'http://www.example.com', ' ', 'http://www.example.com', '']
```

Combine this with `itertools.grouper` and you can iterate over the pairs directly

```python
for plaintext, url in grouper(re.split(pattern, content), 2):
    self.WriteText(plaintext)
    if url:
        self.BeginTextColour(self.url_colour)
        self.BeginUnderline()
        self.BeginURL(url)
        self.WriteText(url)
        self.EndURL()
        self.EndUnderline()
        self.EndTextColour()
```

Reply:
`grouper` is not actually in itertools, it's just a "recipe" in https://docs.python.org/3.7/library/itertools.html#itertools-recipes ; and it needs either `izip_longest` or `zip_longest` (for Python 2 vs 3) which seems to me like a hassle. I hope the updated version is more to your liking.